indico/modules/rb_new/client/js/common/rooms/NonBookablePeriods.jsx

```diff
@@ -0,0 +1,104 @@
+
+import './NonBookablePeriods.module.scss';
+
+
+export default class NonBookablePeriods extends React.Component {
+    static propTypes = {
+        onFocus: PropTypes.func.isRequired,
+        onBlur: PropTypes.func.isRequired,
+        onChange: PropTypes.func.isRequired,
+        value: PropTypes.array.isRequired,
+    };
+
+    handleAddDates = () => {
+        const {value, onChange} = this.props;
+        onChange([
+            ...value,
+            {
+                startDt: serializeDate(moment()),
+                endDt: serializeDate(moment()),
+                key: shortid.generate()
+            }
+        ]);
+        this.setTouched();
+    };
+
+    handleRemoveDates = (key) => {
+        const {value, onChange} = this.props;
+        onChange([...value.filter((dA) => dA.key !== key)]);
+        this.setTouched();
+    };
+
+    handleDatesChange = ({startDate, endDate}, key) => {
+        const {value, onChange} = this.props;
+        onChange([...value.map(v => (v.key === key ? {...v,
+                                                      startDt: serializeDate(startDate),
+                                                      endDt: serializeDate(endDate)} : v))]);
+        this.setTouched();
+    };
+
+    setTouched = () => {
+        // pretend focus+blur to mark the field as touched in case an action changes
+        // the data without actually involving focus and blur of a form element
+        const {onFocus, onBlur} = this.props;
+        onFocus();
+        onBlur();
+    };
+
+    render() {
+        const {value} = this.props;
+        return (
+            <>
+                <Button type="button"
+                        className="room-edit-modal-add-btn"
+                        icon labelPosition="left"
+                        onClick={this.handleAddDates}>
+                    <Icon name="plus" />
+                    <Translate>Add new Nonbookable Periods</Translate>
+                </Button>
+                {value && value.map((dateRangeItem) => {
+                    const {startDt, endDt, key} = dateRangeItem;
+                    return (
+                        <div key={key} className="flex-container">
+                            <DateRangePicker small
+                                             minimumNights={0}
+                                             anchorDirection={ANCHOR_RIGHT}
+                                             startDate={startDt === null ? null : moment(startDt)}
+                                             endDate={endDt === null ? null : moment(endDt)}
+                                             onDatesChange={(dates) => this.handleDatesChange(dates, key)} />
+                            <Icon floated="right" name="trash" className="trash-button" onClick={() => this.handleRemoveDates(key)} />
+                        </div>
+                    );
+                })}
+                {!value && <Translate>No nonbookable periods found</Translate>}
```

Comment:
The same as in `DailyAvailability.jsx`. This message will not be displayed if `value` is an empty array.